### DISCUSSION TITLE
why define an enum in-place in VestingConditionTrigger.schema.json

### DISCUSSION THREAD

**Discussion by haoting-addepar** [Open-Cap-Table-Coalition/Open-Cap-Format-OCF#201] [Q&A] [answered]
All other enums within the schema are defined under `enums` and are referred to via `$ref`. The only exception is that in `VestingConditionTrigger.schema.json`, an enum is defined in place. 

Is there a reason for the exception? I think that we should extract the enum for consistency reasons.

**Comment by pjohnmeyer** [comment, marked as answer]
@haoting-addepar thanks for asking. When I created that, I chose not to extract an enum that was being used in only one place. But, I think there are other places in which we do this extraction (including in vesting). @JSv4 what do you think, should we make an issue for this?

I'd also suggest we add that to our contribution guidelines, if we're going to "always" extract enums.

**Reply by JSv4** [reply]
Thanks for raising, @haoting-addepar. Yeah, @pjohnmeyer, I'd suggest opening an issue. This came up during the Technical Working Group call as well, and I recall the consensus was to structure enums consistently. 

Maybe open two issues here?

1. Update contributors' guidelines as you suggested. 
2. Move the enum in `VestingConditionTrigger.schema.json` to its own enum file.

Anything I'm missing here?

**Reply by haoting-addepar** [reply]
cool. thanks.
